src/Tribe/Cost_Utils.php

```diff
@@ -338,4 +338,78 @@ public function parse_cost_range( $costs, $max_decimals = null ) {
 
 		return (array) $ocost;
 	}
+
+	/**
+	 * @param       string       $original_string_cost A string cost with or without currency symbol,
+	 *                                                 e.g. `10 - 20`, `Free` or `2$ - 4$`.
+	 * @param       array|string $merging_cost         A single string cost representation to merge or a an array of
+	 *                                                 string cost representations to merge, e.g. ['Free', 10, 20,
+	 *                                                 'Donation'] or `Donation`.
+	 * @param       bool         $with_currency_symbol Whether the output should prepend the currency symbol to the
+	 *                                                 numeric costs or not.
+	 * @param array              $sorted_mins          An array of non numeric price minimums sorted smaller to larger,
+	 *                                                 e.g. `['Really free', 'Somewhat free', 'Free with 3 friends']`.
+	 * @param array              $sorted_maxs          An array of non numeric price maximums sorted smaller to larger,
+	 *                                                 e.g. `['Donation min $10', 'Donation min $20', 'Donation min
+	 *                                                 $100']`.
+	 *
+	 * @return string|array The merged cost range.
+	 */
+	public function merge_cost_ranges( $original_string_cost, $merging_cost, $with_currency_symbol, $sorted_mins = array(), $sorted_maxs = array() ) {
+		if ( empty( $merging_cost ) || $original_string_cost === $merging_cost ) {
+			return $original_string_cost;
+
+		}
+
+		$merging_cost               = (array) $merging_cost;
+		$numeric_merging_cost_costs = array();
+		foreach ( $merging_cost as $c ) {
+			if ( is_numeric( $c ) ) {
+				$numeric_merging_cost_costs[] = $c;
+			}
+		}
+
+		$matches = array();
+		preg_match_all( '!\d+(?:\.\d+)?!', $original_string_cost, $matches );
```

Comment:
Given `$10.99 - $59.95` looks like we'd get two matches back - `10.99` and `59.95` - but if the original string cost was `€10,99 - €59,95` we'd get four matches - `10`, `99`, `59` and `95`. Is that a concern or is that handled upstream/elsewhere?